aspnetcore/data/ef-rp/intro/samples/StageSnapShots/cu-part4-migrations/Pages/Students/Delete.cshtml.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.EntityFrameworkCore;
+using ContosoUniversity.Data;
+using ContosoUniversity.Models;
+
+namespace ContosoUniversity.Pages.Students
+{
+    public class DeleteModel : PageModel
+    {
+        private readonly ContosoUniversity.Data.SchoolContext _context;
+
+        public DeleteModel(ContosoUniversity.Data.SchoolContext context)
+        {
+            _context = context;
+        }
+
+        [BindProperty]
+        public Student Student { get; set; }
+
+        public async Task<IActionResult> OnGetAsync(int? id, bool? saveChangesError = false)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            Student = await _context.Students
+                .AsNoTracking()
+                .SingleOrDefaultAsync(m => m.ID == id);
```

Comment:
It's usually best to use FirstOrDefaultAsync unless the code specifically needs to check that there really is no more than one entity returned, because doing that check fetches more data and does more work.